R/importSNV.TCGA.R

```diff
@@ -0,0 +1,156 @@
+
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+
+    #  Validate the vector of MAF file names parameter
+    #  Validate rSNV as an existing file or create new file
+
+    # ==== READ DATA ===========================================================
+
+    # reading data will occur in multiple steps to avoid storing too much data on
+    # the disk drive and cause overload when handeling large files
+
+    # step 0: run terminal comand to gzip all folders and subfolders and remove
+    #         unwanted files like stdout, stdin, manifest etc..
+    # step 1: go through diseases for which data is available and open file
+    # step 2: read lines from file and process data as indicated in "EXTRACT" below
+    # step 3: save processed data to new/merge to rSNV file
+    # step 4: close file
+    # step 5: repeat steps 1-4 for next disease MAF file avilable.
+
+    # ==== "EXTRACT" - REQUIRED DATA ========================================================
+
+    # required data from MAF files for rSNV lie in the columns:
+    # 1 HUGO SYMBOL
+    # 5 chromosome
+    # 6 start
+    # 7 end
+    # 8 strand
+    # 9 Variant_Classification
+    # 10 Variant_type
+    # 11 Reference_allele
+    # 12 Tumor_Seq_Allele1
+    # 13 Tumor_Seq_Allele2
+    # 16 Tumor_Sample_Barcode
+    # 33 Tumor_Sample_UUID
+
+    # read each sample from file and parse the columns listed above and write them
+    # to the rSNV file.
+    # incase of missing value add index of sample entry in rSNV to vector 'missingValues'
+    # when we reach the last sample close file and open new file.
+
+    # ==== "HANDLE" - MISSING VALUES =============================================
+
+    # Incase we encounter missing attributes for samples
+    # Add the sample entry to the rSNV file but record the line number
+    # or the index of that entry.
+    # Make a vector of all entries with missing values
+    # Report all missing entries in the log Below.
+
+    # can also have and option to drop all entries with missing values
+
+    # ====  =================================================
+
+    # ==== WRITE LOG ===========================================================
+
+    if(writeLog) {
+
+        myTitle <- "importSNV.TCGA"
+
+        # Compile function call record
+        myCall <- character()
+        myCall[1] <- "importSNV.TCGA("
+        myCall[2] <- sprintf("rMAF = \"%s\", ", fName)
+        myCall[3] <- sprintf("rSNV = \"%s\", ", net)
+        myCall[4] <- sprintf("silent = %s, ", as.character(silent))
+        myCall[5] <- sprintf("writeLog = %s)", as.character(writeLog))
+        myCall <- paste0(myCall, collapse = "")
+
+        # indicate input object name(s)
+        #   (NA for this importSNV.TCGA())
+
+        # Record progress information
+        myNotes <- character()
+        myNotes <- c(myNotes, sprintf("Files processed - ", nDone))
+        nyNotes <- c(myNotes, sprintf("Files pending - ", nPend))
```

Comment:
I meant pending as in, what MAF files have been processed and what remain from input data
But now that I think about it, it seems irrelevant